### ISSUE TITLE
Add API to resolve configuration object

### ISSUE THREAD

**Issue by muuvmuuv** [stylelint/stylelint#3783]
Prettier has a function to resolve prettier config before actual formatting, I guess stylelint has something like that under the hood too. Would be nice to have this in the API aswell.

In prettier it looks like this:

```js
    const resolveConfig: typeof ResolveConfig = prettier.resolveConfig;
    return resolveConfig(filePath, { useCache: false }).then((config: any) => confif);
```

> https://prettier.io/docs/en/api.html#prettierresolveconfigfilepath-options

I think stylelints "resolveConfig" is here: https://github.com/stylelint/stylelint/blob/0fb8b6d561ca59830cb6d66addf588cfe6169ff1/lib/printConfig.js#L52-L54

**Comment by hudochenkov** [comment]
What is it use case?

**Comment by muuvmuuv** [comment]
I'm working on a new beautifier extension for [Unibeautify](https://github.com/Unibeautify/). 
Unibeautify has a function called [`resolveConfig`](https://github.com/Unibeautify/beautifier-prettier/blob/master/src/index.ts#L49-L59), to retrieve beautifier-specific configuration. 
It also provides the possibility to set options for specific languages for multiple beautifiers (prettier, stylelint...), which will be later transformed into readable breautifier config (e.g. stylelint). The function called [`beautify`](https://github.com/Unibeautify/beautifier-prettier/blob/master/src/index.ts#L60-L89) will then use [this config or the config passed to Unibeautify](https://github.com/Unibeautify/beautifier-prettier/blob/master/src/index.ts#L77-L80) to beautify a file.

**Comment by jeddy3** [comment]
@muuvmuuv Thanks for writing up your use case. 

This is the second request, the other being https://github.com/stylelint/stylelint/issues/3734, to expose some of the internals of stylelint (as part of our API) to the benefit of tool authors. Unless anyone can think of a good reason why not to, I think we should welcome a PR do this (and undertake any necessary refactoring to make it possible) as our internals seem pretty solid and unlikely to change any time soon.

**Comment by jeddy3** [comment]
Triaging issues... 

As no one has raised any objections to this feature, I'm going to label as "help wanted". 

@muuvmuuv Feel free to contribute this feature.

**Comment by is2ei** [comment]
Is this issue still relevant?
It seems https://github.com/stylelint/stylelint/pull/5734 fixed it.
